substrate/state-machine/src/backend.rs

```diff
@@ -66,21 +74,25 @@ impl error::Error for Void {
 /// In-memory backend. Fully recomputes tries on each commit but useful for
 /// tests.
 #[derive(Clone, PartialEq, Eq)]
-pub struct InMemory {
+pub struct InMemory<H, C> {
 	inner: Arc<HashMap<Vec<u8>, Vec<u8>>>,
+	_hasher: PhantomData<H>,
```

Comment:
Curious: why did you opt for an underscore here?

Reply:
No real good reason. I keep going back and forth on what how to name my phantoms, so I end up being inconsistent. Sometimes I like `marker`, sometimes I name it for what it contains, sometimes with a `_` prefix to signal to the reader "never mind this, it's there to make the code compile and is not relevant".

I'd like to have a rule to follow here. What do you use?

Reply:
[Clippy states that underscore-prefixed names](https://rust-lang-nursery.github.io/rust-clippy/master/index.html#used_underscore_binding) are expected to be in there for compile-passing-purposes only and aren't actually used. Which is the case here, and combined with a proper descriptive name, I think this is totally the appropriate way to do it, @dvdplm  :) .

Reply:
"compile-passing-purposes" – rofl